packages/video_player/video_player_android/android/src/test/java/io/flutter/plugins/videoplayer/VideoPlayerTest.java

```diff
@@ -4,12 +4,107 @@
+  private ExoPlayer fakeExoPlayer;
+
+  @Captor private ArgumentCaptor<HashMap<String, Object>> eventCaptor;
+
+  @Before
+  public void before() {
+    MockitoAnnotations.openMocks(this);
+
+    EventChannel fakeEventChannel = mock(EventChannel.class);
+    TextureRegistry.SurfaceTextureEntry fakeSurfaceTextureEntry =
+        mock(TextureRegistry.SurfaceTextureEntry.class);
+    VideoPlayerOptions fakeVideoPlayerOptions = mock(VideoPlayerOptions.class);
+
+    fakeEventSink = mock(QueuingEventSink.class);
+    fakeExoPlayer = mock(ExoPlayer.class);
+    videoPlayer =
+        new VideoPlayer(
+            fakeExoPlayer,
+            fakeEventChannel,
+            fakeSurfaceTextureEntry,
+            fakeVideoPlayerOptions,
+            fakeEventSink);
+  }
+
+  @Test
+  public void sendInitializedSendsExpectedEvent_90and270RotationDegrees() {
+    Format testFormat =
+        new Format.Builder().setWidth(100).setHeight(200).setRotationDegrees(90).build();
+
+    when(fakeExoPlayer.getVideoFormat()).thenReturn(testFormat);
+    when(fakeExoPlayer.getDuration()).thenReturn(10L);
+
+    videoPlayer.isInitialized = true;
+    videoPlayer.sendInitialized();
+
+    verify(fakeEventSink).success(eventCaptor.capture());
+    HashMap<String, Object> event = eventCaptor.getValue();
+
+    assertEquals(event.get("event"), "initialized");
+    assertEquals(event.get("duration"), 10L);
+    assertEquals(event.get("width"), 200);
+    assertEquals(event.get("height"), 100);
+
+    testFormat = new Format.Builder().setWidth(100).setHeight(200).setRotationDegrees(270).build();
+
+    videoPlayer.sendInitialized();
+
+    assertEquals(event.get("event"), "initialized");
+    assertEquals(event.get("duration"), 10L);
+    assertEquals(event.get("width"), 200);
+    assertEquals(event.get("height"), 100);
+  }
+
   @Test
-  public void initPluginDoesNotThrow() {
-    final VideoPlayerPlugin plugin = new VideoPlayerPlugin();
+  public void sendInitializedSendsExpectedEvent_0and180RotationDegrees() {
+    Format testFormat =
+        new Format.Builder().setWidth(100).setHeight(200).setRotationDegrees(0).build();
+
+    when(fakeExoPlayer.getVideoFormat()).thenReturn(testFormat);
+    when(fakeExoPlayer.getDuration()).thenReturn(10L);
+
+    videoPlayer.isInitialized = true;
+    videoPlayer.sendInitialized();
+
+    verify(fakeEventSink).success(eventCaptor.capture());
+    HashMap<String, Object> event = eventCaptor.getValue();
+
+    assertEquals(event.get("event"), "initialized");
+    assertEquals(event.get("duration"), 10L);
+    assertEquals(event.get("width"), 100);
+    assertEquals(event.get("height"), 200);
+
+    testFormat = new Format.Builder().setWidth(100).setHeight(200).setRotationDegrees(180).build();
+
+    videoPlayer.sendInitialized();
+
+    assertEquals(event.get("event"), "initialized");
+    assertEquals(event.get("duration"), 10L);
+    assertEquals(event.get("width"), 100);
+    assertEquals(event.get("height"), 200);
```

Comment:
Here we should also assert `rotationCorrection == 180`. In other cases, we could assert that `rotationCorrection` is not set.